ecommerce/products/templates/product_detail_view.html

```diff
@@ -1,9 +1,14 @@
 <h1>Product</h1>
+<a href="{% url 'home' %}">
+<h3>Home</h3>
+</a>
 <ul>
     <li>
         <div>
             <div><strong>Name: </strong>{{ product.name }}</div>
+            <a href="{% url 'product-by-type' product.product_type %}">
```

Comment:
Indentation is a bit messy here. You should try install a html indentation plugin on pycharm so that it formats all this stuff for you